app/assets/stylesheets/pages/nodes_list.scss

```diff
@@ -0,0 +1,28 @@
+.discovery-control {
```

Comment:
I like the idea, but in this case, I think the `nodes_list` is more like a reusable component across several pages than a page itself.
Maybe for the backend it makes more sense to convert it into its own controller... IDK